plugins/mail-job/impl/src/test/java/org/pentaho/di/job/entries/mail/JobEntryTest.java

```diff
@@ -63,4 +97,44 @@ public void testJobEntrymailPasswordEncrVar() {
     jem.setVariable( "my_pass", "Encrypted 2be98afc86aa7f2e4cb79ce10df81abdc" );
     Assert.assertEquals( jem.getPassword( "${my_pass}" ), "asdf" );
   }
+
+  @Test
+  public void testOauthAuthenticationFailsWithInvalidTokenUrl()  {
+    JobEntryMail jobEntryMail = new JobEntryMail();
+    jobEntryMail.setUsingAuthentication( JobEntryMail.AUTENTICATION_OAUTH );
+    jobEntryMail.setTokenUrl( "http://invalidurl.com/token" );
+    Result result = new Result();
+    jobEntryMail.execute( result, 0 );
+    assertTrue( result.getNrErrors() > 0 );
+  }
+
+  @Test
+  public void testgrantTypeIsClientCredentials() {
+    jobEntryMail.setGrant_type( JobEntryMail.GRANTTYPE_CLIENTCREDENTIALS );
+    assertEquals( JobEntryMail.GRANTTYPE_CLIENTCREDENTIALS, jobEntryMail.getGrant_type() );
+  }
+
+  @Test
+  public void testgrantTypeIsAuthorizationCode() {
+    jobEntryMail.setGrant_type( JobEntryMail.GRANTTYPE_AUTHORIZATION_CODE );
+    assertEquals( JobEntryMail.GRANTTYPE_AUTHORIZATION_CODE, jobEntryMail.getGrant_type() );
+  }
+
+  @Test( expected = RuntimeException.class )
```

Comment:
Same as the comments in the other test; can we be more selective here?  Is there any other check we can do to ensure the test failed in the expected way?